approaches/byte-pair-encoding.md

```diff
@@ -3,5 +3,32 @@ grand_parent: Approaches
 parent: Neural machine translation
 layout: coming_soon
 title: Byte-pair encoding
-description: A data compression algorithm for machine translation
+description: A data compression algorithm for tokenisation for machine translation
 ---
+
+Byte-pair encoding (BPE) is a data compression algorithm where the most common consecutive pair of bytes is replaced by a byte that is not seen in the data.
+It was first introduced in the article titled "_A new Algorithm for Data Compression_" by Philip Gage in the February 1994 edition of C Users Journal.
+
+A very well-known example of byte-pair encoding data compression is, suppose we have a string `aaabdaaabac` which is compressed such that, the most frequent byte pair `aa` is replaced by Z. We now have `ZabdZabac`. 
```

Comment:
To keep content minimal, we can turn this sentence into a step-by-step process. For instance:

```
## Examples

String: `aaabdaaabac`

1- The most frequent byte pair is `aa`, which can be replaced by `Z`. The new string is now `ZabdZabac`.
```